File: youtube-links/src/auth/JwtProvider.ts
import { Jwt, type JwtDto } from "@/shared/features/users/responses/JwtDto";

class JwtProvider {
    getJwtDto(): JwtDto | undefined {
        const tokenString = localStorage.getItem(Jwt.Dto);
        if (tokenString) {
            return JSON.parse(tokenString) as JwtDto;
        }
        return undefined;
    }

    setJwtDto(token: JwtDto | undefined): void {
        if(token != undefined)
            localStorage.setItem(Jwt.Dto, JSON.stringify(token));
    }

    removeJwtDto(): void {
        localStorage.removeItem(Jwt.Dto);
    }
}

export const jwtProvider = new JwtProvider();

File: YoutubeLinks.Blazor/Auth/JwtProvider.cs
using Blazored.LocalStorage;
using YoutubeLinks.Shared.Abstractions;
using YoutubeLinks.Shared.Features.Users.Responses;

namespace YoutubeLinks.Blazor.Auth;

public class JwtProvider(ILocalStorageService localStorageService) : IJwtProvider
{
    public async Task<JwtDto> GetJwtDto()
    {
        return await localStorageService.GetItemAsync<JwtDto>(Jwt.Dto) ?? null;
    }

    public async Task SetJwtDto(JwtDto token)
    {
        await localStorageService.SetItemAsync(Jwt.Dto, token);
    }

    public async Task RemoveJwtDto()
    {
        await localStorageService.RemoveItemAsync(Jwt.Dto);
    }
}